app/Http/Controllers/CourseController.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Carbon\Carbon;
+use App\Models\Course;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Cache;
+use Symfony\Component\HttpFoundation\Response;
+
+
+class CourseController extends Controller
+{   
+    private $courseData;
+
+    public function __construct()
+    {   //database取出來的就是一堆物件，裡面有一個叫做name的屬性，可以直接拿出來用
+        //DB::table('course')->get('name') //原本使用這個操作
+        //但是現在改用模型操作，模型有指定table
+        /**
+         * 減少資料庫查詢的次數，所以在建構子就把需要用到的屬性都放進陣列中
+         * 
+         * 
+         */
+
+        // foreach(Course::all() as $obj){
+        //     $this->courseData[$obj->course_id] = [
+        //         'id' => $obj->id,
+        //         'name' => $obj->name,
+        //         'description' => $obj->description,
+        //         'outline' => $obj->outline, 
+        //         'created_at' => $obj->created_at,
+        //         'updated_at' => $obj->updated_at,
+        //     ];
+        // }
+        //$arr[8]["students"][0]["course_id"]
+        foreach(Course::with('students')->get() as $obj){
+            if(count($obj->students) == 0){
+                $this->courseData[$obj->course_id] = [
+                    'id' => $obj->id,
+                    'name' => $obj->name,
+                    'description' => $obj->description,
+                    'outline' => $obj->outline, 
+                    'created_at' => $obj->created_at,
+                    'updated_at' => $obj->updated_at,
+                    //'students' => $obj->students[0]["first_name"]." ".$obj->students[0]["last_name"],
+                    'students' => "Seems that nobody wanna take this QQ",
+                ];
+            } else {
+                $this->courseData[$obj->course_id] = [
+                    'id' => $obj->id,
+                    'name' => $obj->name,
+                    'description' => $obj->description,
+                    'outline' => $obj->outline, 
+                    'created_at' => $obj->created_at,
+                    'updated_at' => $obj->updated_at,
+                    'students' => $obj->students,
+                ];
+                
+            }
+            
+        }
+        
+    }
+    /**
+     * 判斷網址/course後面接什麼
+     * 
+     * if null return courseIndex.blade.php
+     * if not isset return HTTP_NOT_FOUND
+     * if true return course.blade.php
+     * @param $request
+     * 
+     * @return courseIndex | course | HTTP_NOT_FOUNT
+     */
+    public function index(Request $request)
```

Comment:
```suggestion
    public function index(Request $request, $courseAddr = null)
```
可以直接取 route rule 的變數